src/components/MainPage/index.js

```diff
@@ -0,0 +1,211 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types'
+import { loadAllUsers } from './../../actions';
+
+import decoratorReset from './../../decorators/toggleReset';
+import { mapToArr } from './../../changeDataStructure';
+import './style.scss';
+
+class UserTable extends Component {
+	static propTypes = {
+		users: PropTypes.array.isRequired,		
+		toggleReset: PropTypes.func
+	}
+
+	constructor(props) {
+		super(props);
+		debugger;
+		this.state = {
+			sortByName: false,
+			sortByAge: false,
+			userName: '',
+			currentUser: null
+		};
+
+		this.handleChange = this.handleChange.bind(this);
+		this.handleChangeUser = this.handleChangeUser.bind(this);
+		this.handleSortByName = this.handleSortByName.bind(this);
+		this.handleSortByAge = this.handleSortByAge.bind(this);
+		this.handleReset = this.handleReset.bind(this);
+	}
+
+	handleChange(item) {
+		
+		this.setState({
+			currentUser: item
+		});
+	}
+	handleSortByName(ev) {
+
+		this.setState({
+			sortByName: true,
+			sortByAge: false,
+			currentUser: null
+		});
+	}
+	handleSortByAge(ev) {
+		this.setState({
+			sortByName: false,
+			sortByAge: true,
+			currentUser: null
+
+		});
+	}
+	handleReset(ev) {
+		this.setState({
+			sortByName: false,
+			sortByAge: false,
+			currentUser: null
+
+		});
+		let { toggleReset } = this.props;
+		toggleReset(); // reset in decorator
```

Comment:
Why do we need HOC toggleSort? Is it re-used anywhere else?

Reply:
That needs to change a parent state and get previous list users

Reply:
@BondarenkoMaks thanks for your explanation! I understand what `toggleReset` is doing. However, my question concerns using HOC pattern here.